integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java

```diff
@@ -103,136 +129,276 @@
+
+      patch = new V1Patch(patchStr);
+      assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+          "patchDomainCustomResource failed");
+
+      logger.info("verify domain changed event is logged");
+      checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_CHANGED, "Normal", timestamp);
+    }
+  }
 
-    //verify domain processing retrying event
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_PROCESSING_RETRYING,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_PROCESSING_RETRYING, "Normal", timestamp));
+  /**
+   * Test verifies there is only 1 DomainProcessing Starting/Completed event is logged in a multi cluster domain.
+   */
+  @Order(5)
+  @Test
+  public void testK8SEventsMultiClusterEvents() {
+    createNewCluster();
+    DateTime timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    scaleClusterWithRestApi(domainUid, cluster2Name, 1,
+        externalRestHttpsPort, opNamespace, opServiceAccount);
+    logger.info("verify the DomainProcessing Starting/Completed event is generated");
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_PROCESSING_STARTING, "Normal", timestamp);
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_PROCESSING_COMPLETED, "Normal", timestamp);
+    logger.info("verify the only 1 DomainProcessing Starting/Completed event is generated");
+    assertEquals(1, getEventCount(domainNamespace1, domainUid, DOMAIN_PROCESSING_STARTING, timestamp));
+    assertEquals(1, getEventCount(domainNamespace1, domainUid, DOMAIN_PROCESSING_COMPLETED, timestamp));
+  }
 
-    //verify domain processing aborted event
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_PROCESSING_ABORTED,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_PROCESSING_ABORTED, "Warning", timestamp));
+  /**
+   * Scale the cluster beyond maximum dynamic cluster size and verify the warning event is generated.
+   */
+  @Order(6)
+  @Test
+  @Disabled("Bug - OWLS-87019")
+  public void testDomainK8sEventsScalePastMax() {
+    DateTime timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
+    scaleClusterWithRestApi(domainUid, cluster1Name, 3, externalRestHttpsPort, opNamespace, opServiceAccount);
```

Comment:
Can you try scaling cluster by patching the domain crd? I think there are some validation checking in REST endpoint and thus the domain would not get updated if the replica count > max cluster size.

Reply:
Yes with patching the domain I can see the event. Does it mean the test needs to be modified to scale with patching?